pennylane/fermi/fermionic.py

```diff
@@ -123,7 +123,7 @@ def __str__(self):
 
     def __repr__(self):
         r"""Terminal representation of a FermiWord"""
-        return str(self)
+        return str(self.sorted_dic)
```

Comment:
```suggestion
        return f"FermiWord({self.sorted_dic})"
```
I think this is more informative otherwise, we lose information regarding `FermiWord` when it is used in the `repr` for `FermiSentence`.